python/event_generator.cpp

```diff
@@ -0,0 +1,137 @@
+            "Events in the regular sequence stop being delivered after this time (in ms).")
+        .def_readwrite("dt", &regular_schedule_shim::dt,
+            "The distance between time points in the regular sequence (in ms).")
+        .def("__str__", &schedule_regular_string)
+        .def("__repr__",&schedule_regular_string);
+
+    // Explicit schedule
+    pybind11::class_<explicit_schedule_shim> explicit_schedule(m, "explicit_schedule",
+        "Describes an explicit schedule at times given explicitly via a provided sorted sequence.");
+
+    explicit_schedule
+        .def(pybind11::init<>(), "Construct an explicit schedule with an empty list of times.")
+        .def(pybind11::init<pybind11::list>(),
+            "times"_a,
+            "Construct an explicit schedule with argument:\n"
+            "  times: A list of times in the explicit schedule (in ms).")
+        .def_readwrite("times", &explicit_schedule_shim::py_times,
+            "A list of times in the explicit schedule (in ms).")
+        .def("__str__", &schedule_explicit_string)
+        .def("__repr__",&schedule_explicit_string);
+
+    // Poisson schedule
+    pybind11::class_<poisson_schedule_shim> poisson_schedule(m, "poisson_schedule",
+        "Describes a schedule at Poisson point process with rate 1/mean_dt, restricted to non-negative times.");
+
+    poisson_schedule
+        .def(pybind11::init<>(),
+            "Construct a Poisson schedule with default arguments:\n"
+            "  tstart: 0 ms.\n"
+            "  freq:   10 Hz.\n"
+            "  seed:   0 for a Mersenne Twister pseudo-random generator of 64-bit numbers with a state size of 19937 bits.")
+        .def(pybind11::init<arb::time_type, std::mt19937_64::result_type>(),
+            "freq"_a, "seed"_a,
+            "Construct a Poisson schedule with arguments:\n"
+            "  freq: The expected frequency (in Hz).\n"
+            "  seed: The seed of the Mersenne Twister pseudo-random generator of 64-bit numbers with a state size of 19937 bits.")
+        .def(pybind11::init<arb::time_type, arb::time_type, std::mt19937_64::result_type>(),
+            "tstart"_a, "freq"_a, "seed"_a,
+            "Construct a Poisson schedule with arguments:\n"
+            "  tstart: Events in the Poisson schedule start being delivered from this time(in ms).\n"
+            "  freq:   The expected frequency (in Hz).\n"
+            "  seed:   The seed of the Mersenne Twister pseudo-random generator of 64-bit numbers with a state size of 19937 bits.")
+        .def_readwrite("tstart", &poisson_schedule_shim::tstart,
+            "Events in the Poisson schedule start being delivered from this time (in ms).")
+        .def_readwrite("freq", &poisson_schedule_shim::freq,
+            "The expected frequency (in Hz).")
+        .def_readwrite("seed", &poisson_schedule_shim::seed,
+            "The seed of the Mersenne Twister pseudo-random generator of 64-bit numbers with a state size of 19937 bits.")
```

Comment:
Simplify to "Set seed for the random number generator."

Reply:
Done (leaving out "Set" due to consistancy).